### DISCUSSION TITLE
Controlled Carousel

### DISCUSSION THREAD

**Discussion by MoSattler** [wix/react-native-ui-lib#1890] [Ideas]
I would like to be able to programmatically set the current page of a carousel. E.g. if somebody clicks on a button, I want to scroll to a certain page of the carrousel. In other words, I would like to be able to have a controlled carousel.

**Comment by ethanshar** [comment]
Hi @MoSattler 
have you considered passing a ref to the carousel component and invoke its `goToPage` method? 

```

const carouselRef = useRef();

const goToPage = useCallback((pageIndex) => {
  carouselRef.current.goToPage(pageIndex);
}, [])


return <Carouse ref={carouselRef} ... />

```

**Reply by MoSattler** [reply]
It's been a while now, so nut sure, but I believe I tried that, and it did not work as I imaged.
